### PR TITLE
fix: conditionnal version load

### PR DESCRIPTION
**Pull request by zekth**
fix the issue introduced with #2605 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by kibertoad — APPROVED** [review]

File: fastify.js [hunk 1]
```diff
@@ -595,12 +595,14 @@ function loadVersion () {
...
   const fs = require('fs')
   const path = require('path')
-  const pkgPath = path.join(__dirname, 'package.json')
-  if (fs.accessSync(pkgPath, fs.constants.R_OK)) {
+  try {
+    const pkgPath = path.join(__dirname, 'package.json')
+    fs.accessSync(pkgPath, fs.constants.R_OK)
     const pkg = JSON.parse(fs.readFileSync(pkgPath))
-    return pkg.version
+    return pkg.name === 'fastify' ? pkg.version : undefined
```
Comment (mcollina) [inline fastify.js:602]: Can you please add a test that verifies this condition?

**Comment by agramian** [comment]
@zekth this looks good and what I was planning to do.

The only other consideration is that in `checkVersion` the `undefined` value for `version` must be handled to avoid the error I mentioned in https://github.com/fastify/fastify/pull/2605#discussion_r501179125.

Something like `if (!this.version) return` can be added [here](https://github.com/fastify/fastify/blob/master/lib/pluginUtils.js#L89-L97).

File: test/internals/version.test.js [hunk 2]
```diff
@@ -3,10 +3,17 @@
 const t = require('tap')
 const test = t.test
 const proxyquire = require('proxyquire')
-const Fastify = proxyquire('../..', { fs: { accessSync: () => false } })
+const Fastify = proxyquire('../..', { fs: { accessSync: () => { throw Error('error') } } })
+const Fastify2 = proxyquire('../..', { fs: { accessSync: () => { }, readFileSync: () => JSON.stringify({ name: 'foo', version: '6.6.6' }) } })
```
Comment (mcollina) [inline test/internals/version.test.js:7]: Could you please move the proxyquire calls in the test themselves?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Pull request merged**